source/WebConfig/cosa_webconfig_internal.c

```diff
@@ -178,6 +181,94 @@ int Set_Webconfig_URL( char *pString)
         return 1;
 }
 
+int Get_Supplementary_URL( char *name, char *pString)
+{
+    PCOSA_DATAMODEL_WEBCONFIG            pMyObject           = (PCOSA_DATAMODEL_WEBCONFIG)g_pCosaBEManager->hWebConfig;
+    WebcfgDebug("-------- %s ----- Enter-- ---\n",__FUNCTION__);
+
+        if((pMyObject != NULL) &&  (strlen(pMyObject->Telemetry)>0) && (strncmp(name, "Telemetry",strlen(name)+1)) == 0)
+        {
+		WebcfgDebug("pMyObject->Telemetry %s\n", pMyObject->Telemetry);
+                WebcfgDebug("%s ----- updating pString ------\n",__FUNCTION__);
+
+		AnscCopyString( pString,pMyObject->Telemetry );
+		WebcfgDebug("pString %s\n",pString);
+        }
+        else
+        {
+		char *tempDBUrl = NULL;
+                int   retPsmGet    = CCSP_SUCCESS;
+		char *tempParam = (char *) malloc (sizeof(char)*MAX_BUFF_SIZE);
+
+                snprintf(tempParam, MAX_BUFF_SIZE, "%s%s", WEBCONFIG_PARAM_SUPPLEMENTARY_SERVICE, name);
+                WebcfgDebug("tempParam is %s\n", tempParam);
+                retPsmGet = PSM_Get_Record_Value2(bus_handle,g_Subsystem, tempParam, NULL, &tempDBUrl);
+		WebcfgDebug("tempDBUrl is %s\n", tempDBUrl);
+                if (retPsmGet == CCSP_SUCCESS)
+                {
+			WebcfgDebug("retPsmGet success\n");
+			AnscCopyString( pString,tempDBUrl);
+                        WebcfgDebug("pString %s\n",pString);
+			free(tempParam);
+                }
+                else
+                {
+                        WebcfgError("psm_get failed ret %d for parameter %s%s\n", retPsmGet, WEBCONFIG_PARAM_SUPPLEMENTARY_SERVICE, name);
+			free(tempParam);
+                        return 0;
+                }
+        }
+
+        WebcfgDebug("-------- %s ----- Exit ------\n",__FUNCTION__);
+        return 1;
+}
+
+int Set_Supplementary_URL( char *name, char *pString)
+{
+    PCOSA_DATAMODEL_WEBCONFIG            pMyObject           = (PCOSA_DATAMODEL_WEBCONFIG)g_pCosaBEManager->hWebConfig;
+    WebcfgDebug("-------- %s ----- Enter ------\n",__FUNCTION__);
+    int retPsmSet = CCSP_FAILURE;
+    char *tempParam = (char *) malloc (sizeof(char)*MAX_BUFF_SIZE);
+
+        if ((strncasecmp(name, "Telemetry",strlen(name)+1)) == 0)
+        {
+                memset( pMyObject->Telemetry, 0, sizeof( pMyObject->Telemetry ));
+                AnscCopyString( pMyObject->Telemetry, pString );
+                snprintf(tempParam, MAX_BUFF_SIZE, "%s%s", WEBCONFIG_PARAM_SUPPLEMENTARY_SERVICE, name);
+		WebcfgDebug("tempParam is %s\n", tempParam);
+		retPsmSet = PSM_Set_Record_Value2(bus_handle,g_Subsystem, tempParam, ccsp_string, pString);
+        }
+        else
+        {
+                WebcfgError("Invalid supplementary doc name\n");
```

Comment:
"tempParam" is not freed in this failure case, avoid it by moving it inside as mentioned in above comment.